tests/Functional/Driver/PDO/SQLSrv/DriverTest.php

```diff
@@ -77,4 +78,31 @@ public function testDriverOptions(): void
                 ->getAttribute(PDO::ATTR_CASE)
         );
     }
+
+    /**
+     * @dataProvider stringDataProvider
+     */
+    public function testDriverStringBinding(string $bindingType, string $expectedSqlType): void
+    {
+        $statement = $this->connection->prepare(
+            'SELECT sql_variant_property(:parameter, \'BaseType\') AS type'
+        );
+
+        $statement->bindValue(':parameter', 'TEST', $bindingType);
+        $results = $statement->execute()->fetchAssociative();
+
+        self::assertNotFalse($results);
+        self::assertEquals($expectedSqlType, $results['type']);
+    }
+
+    /**
+     * @return array<int, array<int, string>>
+     */
+    public static function stringDataProvider(): array
+    {
+        return [
+            [Types::VARCHAR, 'varchar'],
```

Comment:
I'd rather use the parameter types here (`ParameterType::STRING` and `ParameterType::ASCII_STRING`). DBAL types is a higher-level abstraction that doesn't need to be tested at the driver level.